tests/contrib/anthropic/test_anthropic.py

```diff
@@ -0,0 +1,401 @@
+
+
+@pytest.mark.snapshot(token="tests.contrib.anthropic.test_anthropic.test_anthropic_llm_stream", ignores=["resource"])
+def test_anthropic_llm_sync_stream(anthropic, request_vcr):
+    llm = anthropic.Anthropic()
+    with request_vcr.use_cassette("anthropic_completion_stream.yaml"):
+        stream = llm.messages.create(
+            model="claude-3-opus-20240229",
+            max_tokens=15,
+            messages=[
+                {
+                    "role": "user",
+                    "content": [
+                        {
+                            "type": "text",
+                            "text": "Can you explain what Descartes meant by 'I think, therefore I am'?",
+                        }
+                    ],
+                },
+            ],
+            stream=True,
+        )
+        for _ in stream:
+            pass
+
+
+@pytest.mark.snapshot(token="tests.contrib.anthropic.test_anthropic.test_anthropic_llm_tools", ignores=["resource"])
+@pytest.mark.skipif(ANTHROPIC_VERSION < (0, 27), reason="Anthropic Tools not available until 0.27.0, skipping.")
+def test_anthropic_llm_sync_tools(anthropic, request_vcr):
+    llm = anthropic.Anthropic()
+    with request_vcr.use_cassette("anthropic_completion_tools_part_1.yaml"):
+        message = llm.messages.create(
+            model="claude-3-opus-20240229",
+            max_tokens=200,
+            messages=[{"role": "user", "content": "What is the result of 1,984,135 * 9,343,116?"}],
+            tools=tools,
+        )
+
+    with request_vcr.use_cassette("anthropic_completion_tools_part_2.yaml"):
+        if message.stop_reason == "tool_use":
+            tool_use = next(block for block in message.content if block.type == "tool_use")
+            tool_name = tool_use.name
+            tool_input = tool_use.input
+
+            tool_result = process_tool_call(tool_name, tool_input)
+
+            response = llm.messages.create(
+                model="claude-3-opus-20240229",
+                max_tokens=500,
+                messages=[
+                    {"role": "user", "content": "What is the result of 1,984,135 * 9,343,116?"},
+                    {"role": "assistant", "content": message.content},
+                    {
+                        "role": "user",
+                        "content": [
+                            {
+                                "type": "tool_result",
+                                "tool_use_id": tool_use.id,
+                                "content": tool_result,
+                            }
+                        ],
+                    },
+                ],
+                tools=tools,
+            )
+        else:
+            response = message
+
+        final_response = next(
+            (block.text for block in response.content if hasattr(block, "text")),
+            None,
+        )
+        assert final_response is not None
```

Comment:
I don't think we need to make the test this complicated (2 cassettes, 2 message calls, running an actual tool function, etc). We just need to test that we tag the correct values if we specify tool calling in a message call.

Can we just reproduce the first basic example in anthropic's tool call [docs](https://docs.anthropic.com/en/docs/tool-use-examples#single-tool)?

Reply:
This is the full example from their demo examples. Honestly I think we should keep it, since tools use will result in different context blocks. It doesn't really hurt to keep it

Reply:
done